networkx/algorithms/dag.py

```diff
@@ -170,10 +180,12 @@ def topological_sort(G):
 
     indegree_map = {v: d for v, d in G.in_degree() if d > 0}
     # These nodes have zero indegree and ready to be returned.
-    zero_indegree = [v for v, d in G.in_degree() if d == 0]
+    zero_indegree = deque(v for v, d in G.in_degree() if d == 0)
 
+    generation = 0
+    last_node_in_generation = zero_indegree[-1]
```

Comment:
I believe this line is responsible for the test failures - this will raise `IndexError`s if `zero_indegree` is empty.

Reply:
Thanks for following up @rossbar. Updated. If the zero-indegree is empty I decided to have it raise a separate explicit error from the one at the bottom, since we know that this is specifically due to a cycle in the graph, and not from the graph changing during iteration.